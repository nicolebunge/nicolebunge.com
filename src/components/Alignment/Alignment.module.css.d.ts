declare const styles: {
  readonly alignment: string;
  readonly 'alignment-align-center': string;
  readonly 'alignment-align-end': string;
  readonly 'alignment-align-start': string;
  readonly 'alignment-justify-center': string;
  readonly 'alignment-justify-end': string;
  readonly 'alignment-justify-space-between': string;
  readonly 'alignment-justify-start': string;
};
export = styles;
