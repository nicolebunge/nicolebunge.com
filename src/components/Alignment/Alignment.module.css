@media (width >= 640px) {
  .alignment {
    display: flex;
  }
}

.alignment--align-center {
  align-items: center;
}

.alignment--align-end {
  align-items: end;
}

.alignment--align-start {
  align-items: start;
}

.alignment--justify-center {
  justify-content: center;
}

.alignment--justify-end {
  justify-content: end;
}

.alignment--justify-space-between {
  justify-content: space-between;
}

.alignment--justify-start {
  justify-content: start;
}
