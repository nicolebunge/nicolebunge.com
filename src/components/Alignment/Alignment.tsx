/* eslint-disable @typescript-eslint/no-non-null-assertion */

import clsx from 'clsx';
import React, { HTMLAttributes, ReactNode } from 'react';
import {
  alignment,
  alignmentAlignCenter,
  alignmentAlignEnd,
  alignmentAlignStart,
  alignmentJustifyCenter,
  alignmentJustifyEnd,
  alignmentJustifySpaceBetween,
  alignmentJustifyStart,
} from './Alignment.module.css';

type Align = 'center' | 'end' | 'start';
type Justify = 'center' | 'end' | 'space-between' | 'start';

export interface AlignmentProps extends HTMLAttributes<HTMLDivElement> {
  align?: Align;
  children: ReactNode;
  justify?: Justify;
}

const alignClassNames = {
  center: alignmentAlignCenter,
  end: alignmentAlignEnd,
  start: alignmentAlignStart,
};

const justifyClassNames = {
  center: alignmentJustifyCenter,
  end: alignmentJustifyEnd,
  'space-between': alignmentJustifySpaceBetween,
  start: alignmentJustifyStart,
};

function Alignment(props: AlignmentProps): JSX.Element {
  const { align, className, justify, ...otherProps } = props;

  return (
    <div
      className={clsx(className, {
        [alignment]: align || justify,
        [alignClassNames[align!]]: align,
        [justifyClassNames[justify!]]: justify,
      })}
      {...otherProps}
    />
  );
}

export default Alignment;
