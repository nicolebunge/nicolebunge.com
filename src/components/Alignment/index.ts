export { default } from './Alignment';
export * from './Alignment';
