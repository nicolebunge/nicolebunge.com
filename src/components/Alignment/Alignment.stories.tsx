import Alignment from '.';

export default {
  component: Alignment,
};

export const Default = {
  args: {
    children: 'Alignment',
  },
};

export const AlignCenter = {
  args: {
    align: 'center',
  },
};

export const JustifyCenter = {
  args: {
    justify: 'center',
  },
};

export const Center = {
  args: {
    align: 'center',
    justify: 'center',
  },
};
