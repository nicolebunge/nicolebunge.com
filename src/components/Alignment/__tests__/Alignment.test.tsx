import { composeStories } from '@storybook/testing-react';
import { render } from '@testing-library/react';
import React from 'react';
import * as stories from '../Alignment.stories';

const { Default } = composeStories(stories);

describe('Alignment component', () => {
  it('should render correctly', () => {
    const { container } = render(<Default />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
