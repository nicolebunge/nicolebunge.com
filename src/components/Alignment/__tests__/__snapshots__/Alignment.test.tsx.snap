// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alignment component should render correctly 1`] = `
<div
  class=""
>
  Alignment
</div>
`;
